cg/apps/sequencing_metrics_parser/parsers/bcl_convert.py

```diff
@@ -0,0 +1,204 @@
+            file_format=FileFormat.CSV, file_path=self.adapter_metrics_path, read_to_dict=True
+        )
+        for row in metrics_reader:
+            parsed_metrics.append(
+                BclConvertAdapterMetrics(
+                    lane=int(row["Lane"]),
+                    sample_internal_id=row["Sample_ID"],
+                    sample_project=row["Sample_Project"],
+                    read_number=row["ReadNumber"],
+                    sample_bases=row["SampleBases"],
+                )
+            )
+        return parsed_metrics
+
+    def get_nr_of_header_lines_in_sample_sheet(
+        self,
+    ) -> int:
+        """Return the number of header lines in a sample sheet.
+        Any lines before and including the line starting with [Data] is considered the header."""
+        csv_reader = ReadFile.get_content_from_file(FileFormat.CSV, self.sample_sheet_path)
+        header_line_count: int = 1
+        for line in csv_reader:
+            if SampleSheetHeaderColumnNames.DATA.value in line:
+                break
+            header_line_count += 1
+        return header_line_count
+
+    def parse_sample_sheet_file(self):
+        """Read in a sample sheet starting from the SAMPLE_SHEET_DATA_HEADER."""
+        LOG.info(f"Parsing BCLConvert sample sheet file: {self.sample_sheet_path}")
+        header_line_count: int = self.get_nr_of_header_lines_in_sample_sheet()
+        sample_sheet: List[BclConvertSampleSheet] = []
+        with open(self.sample_sheet_path, "r") as sample_sheet_file:
+            for _ in range(header_line_count):
+                next(sample_sheet_file)
+            reader = csv.DictReader(sample_sheet_file)
+            for row in reader:
+                sample_sheet.append(
+                    BclConvertSampleSheet(
+                        flow_cell_name=row["FCID"],
+                        lane=row["Lane"],
+                        sample_internal_id=row["Sample_ID"],
+                        sample_name=row["SampleName"],
+                        control=row["Control"],
+                        sample_project=row["Sample_Project"],
+                    )
+                )
+        return sample_sheet
+
+    def parse_run_info_file(self) -> BclConvertRunInfo:
+        LOG.info(f"Parsing Run info XML {self.run_info_path}")
+        parsed_metrics = BclConvertRunInfo(tree=ET.parse(self.run_info_path))
+        return parsed_metrics
+
+    def get_metrics_for_sample_internal_id(
+        self, sample_internal_id: str, metrics: List
+    ) -> List[
+        Union[
+            BclConvertQualityMetrics,
+            BclConvertDemuxMetrics,
+            BclConvertAdapterMetrics,
+            BclConvertSampleSheet,
+        ]
+    ]:
+        """Return all quality metrics for a given sample internal id."""
+        return [metric for metric in metrics if metric.sample_internal_id == sample_internal_id]
+
+    def _calculate_total_read_counts_for_sample_internal_id(
```

Comment:
Right now I do not understand how these functions will be used (it is a draft I know). If they are specif to model and can be calculated inside - I think it would be good. Then the return would be stored as an attribute on the model

Reply:
Yes I am not 100% sure how to handle the calculations, I decided to do that in a separate PR for now.